File: c#/VerticeLibTests/Strings/StringsTest.cs
namespace VerticeLibTests.Strings
{
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class StringsTest
    {
        [TestMethod]
        public void TestAfter()
        {
            string test1 = "A = B ";

            // Must return string after delimiter = without trimming.
            Assert.AreEqual(" B ", VerticeLib.Strings.Strings.After(test1, "="));

            // Must return trimmed string after delimiter =
            Assert.AreEqual("B", VerticeLib.Strings.Strings.After(test1, "=", true));

            // Must return string after first delimiter is found.
            Assert.AreEqual("= B ", VerticeLib.Strings.Strings.After(test1, " "));

            // Must return trimmed string after first delimiter is found.
            Assert.AreEqual("= B", VerticeLib.Strings.Strings.After(test1, " ", true));

            // Must return string.Empty if delimiter is not found
            Assert.AreEqual(string.Empty, VerticeLib.Strings.Strings.After(test1, "?"));
            Assert.AreEqual(string.Empty, VerticeLib.Strings.Strings.After(test1, "?", true));

            // Compounded delimiter: Must return string after first delimiter is found.
            Assert.AreEqual(" ", VerticeLib.Strings.Strings.After(test1, "= B"));

            // Compounded delimiter: Must return trimmed string after first delimiter is found.
            Assert.AreEqual(string.Empty, VerticeLib.Strings.Strings.After(test1, "= B", true));

            // Working with multi-line text.
            string test2 = @"A

 = B ";

            Assert.AreEqual(" B ", VerticeLib.Strings.Strings.After(test2, "="));
            Assert.AreEqual("B", VerticeLib.Strings.Strings.After(test2, "=", true));
            Assert.AreEqual(" = B ", VerticeLib.Strings.Strings.After(test2, System.Environment.NewLine + System.Environment.NewLine));
            Assert.AreEqual("= B", VerticeLib.Strings.Strings.After(test2, System.Environment.NewLine + System.Environment.NewLine, true));
        }
    }
}

File: js/test/Strings/Strings.test.js
const Strings = require('./../../app/Strings/Strings');

describe("Unit Tests for Strings Class", () => {
    test('1) After tests', () => {
        const test1 = "A = B ";

        // Must return string after delimiter = without trimming.
        expect(Strings.After(test1, "=")).toBe(" B ");

        // Must return trimmed string after delimiter =
        expect(Strings.After(test1, "=", true)).toBe("B");

        // Must return string after first delimiter is found.
        expect(Strings.After(test1, " ")).toBe("= B ");

        // Must return trimmed string after first delimiter is found.
        expect(Strings.After(test1, " ", true)).toBe("= B");

        // Must return "" if delimiter is not found
        expect(Strings.After(test1, "?")).toBe("");
        expect(Strings.After(test1, "?", true)).toBe("");

        // Compounded delimiter: Must return string after first delimiter is found.
        expect(Strings.After(test1, "= B")).toBe(" ");

        // Compounded delimiter: Must return trimmed string after first delimiter is found.
        expect(Strings.After(test1, "= B", true)).toBe("");

        // Working with multi-line text.
        const test2 = "A\n\n = B ";

        expect(Strings.After(test2, "=")).toBe(" B ");
        expect(Strings.After(test2, "=", true)).toBe("B");
        expect(Strings.After(test2, "\n\n")).toBe(" = B ");
        expect(Strings.After(test2, "\n\n", true)).toBe("= B");
    });
})
